### PR TITLE
[core] telemetry for oom occurences

### PR DESCRIPTION
**Pull request by clarng**
Signed-off-by: Clarence Ng <clarence.wyng@gmail.com>

## Why are these changes needed?

track how often oom triggers in the wild

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/_private/usage/usage_lib.py
Comment (jjyao) [inline python/ray/_private/usage/usage_lib.py:270]: Could you apply these two lines of change to make sure TagKeys are in-sync.

File: python/ray/_private/usage/usage_lib.py
Comment (jjyao) [inline python/ray/_private/usage/usage_lib.py:266]: Could you document what the values are?

**Comment by jjyao** [comment]
cc @thomasdesr for security review.

File: python/ray/_private/usage/usage_lib.py
Comment (jjyao) [inline python/ray/_private/usage/usage_lib.py:267]: Could you also comment where those two are emitted?

File: python/ray/tests/test_usage_stats.py
Comment (jjyao) [inline python/ray/tests/test_usage_stats.py:219]: is it flaky, should we use `wait_for_condition`?

**Review by jjyao — APPROVED** [review]
LGTM. I'll follow up with https://github.com/ray-project/ray/pull/30427.

cc @iycheng for the gcs side change.

**Review by ericl — APPROVED** [review]

**Comment by ericl** [comment]
Will defer to @iycheng for merge.

**Comment by clarng** [comment]
> I feel we shouldn't use kv instance here directly for stats. We should use general APIs provided by GCS (GlobalStateAcessor/GcsClient). I'm thinking about Raylet/CoreWorker which want to store the stats and we probably don't want to have different approaches to record the data.
> 
> There is protocol for telemetry and now we have already got python version where core logical is written in py. I'd think in the future, it's going to be very hard to maintain (the same protocol in py/java/gcs/coreworker/raylet).

We added a TODO to do the the way you suggested (gcs client). The reason it is a TODO right now to integrate with gcs client is because it is crashing / hang when we try to use it directly - we are not sure if it is because we are calling gcs from gcs. Jiajun was looking into this last Friday.

The hope is to get something functional and migrate to the gcs client approach as soon as we figured out what was going on that prevented it.

**Comment by iycheng** [comment]
@clarng  that sgtm then! Thanks for the explanation!

**Comment by clarng** [comment]
> Lmk when tests pass!

Looks like it is passing now, some failures look unrelated

**Pull request merged**
